lib/msf/util/exe.rb

```diff
@@ -854,6 +854,12 @@ def self.to_osx_app(exe, opts = {})
     hidden      = opts.fetch(:hidden, true)
     plist_extra = opts.fetch(:plist_extra, '')
 
+    # The app won't execute if the macho isn't executable
+    macho = File.new(exe_name, 'w')
+    macho.write(exe)
+    macho.close
+    FileUtils.chmod(777, macho)
```

Comment:
Note value would need to be ```FileUtils.chmod(0777, macho)``` || ```FileUtils.chmod(0755, macho)```
to make this file executable as all 4 values are required to get proper chmod.